src/app/user-event/assessment-form.component.ts

```diff
@@ -405,12 +408,14 @@ export class AssessmentFormComponent implements OnInit, OnChanges {
     });
   }
 
-  protected _parseUnsavedAnswers(unsavedAnswers: string) {
+  protected _parseUnsavedAssessment(unsavedAssessment: string) {
     try {
-      const parsedAnswers = JSON.parse(unsavedAnswers);
-      if (!!parsedAnswers.length) {
-        this.answers = parsedAnswers;
+      const parsedAssessment: AssessmentModel = JSON.parse(unsavedAssessment);
```

Comment:
I'm not quitely sure that is correct, because `AssessmentModel` is class and variable that are typed should be a instance of it, but currently construction more look likes work with interface.
I suggest to fix it in that way:
```
const parsedAssessment: AssessmentModel = new AssessmentModel(JSON.parse(unsavedAssessment));
```